portfolio/src/main/java/com/google/sps/servlets/PizzaSliceDataServlet.java

```diff
@@ -0,0 +1,104 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.sps.data.User;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.LinkedHashMap;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Scanner;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+
+@WebServlet("/pizza-data")
+public class PizzaSliceDataServlet extends HttpServlet {
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("application/json");
+    Query query = new Query("Vote").addSort("slice", SortDirection.ASCENDING);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    Map<Integer, Integer> pizzaVotes = new LinkedHashMap<>();
+    pizzaVotes.put(0, 0);
+    pizzaVotes.put(1, 0);
+    pizzaVotes.put(2, 0);
+    pizzaVotes.put(3, 0);
+    pizzaVotes.put(4, 0);
+    pizzaVotes.put(5, 0);
+    for (Entity entity : results.asIterable()) {
+      Integer slice = (int) (long) entity.getProperty("slice");
+      pizzaVotes.put(slice, pizzaVotes.get(slice) + 1);
+    }
```

Comment:
This looks dangerous. If there is a slice value in one entity that is not in [0,...,5] you get a null pointer back from the get() method.
Maybe better use https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#getOrDefault-java.lang.Object-V- with default 0 and remove the lines where you initialize the hash map. 

Another question: Why are you using a hashmap instead of a simple array if you expect the slice values to be in [0,..,5]?

Reply:
Yes, it's the good question why -- I'll change. 
What about entity: could I somehow guarantee that values is in [0, 5]? When I put in to datastore, for example. Or should I also check here that the value is in interval?

Reply:
You should check it here too. Maybe you change the code for putting it into the datastore or there is a bug in that code then it would also break this here.